src/commcare_cloud/ansible/roles/formplayer/defaults/main.yml

```diff
@@ -3,4 +3,5 @@ formplayer_sentry_dsn: ''
 # (you can't use s for seconds, but m, h, d work for minute, hour, day)
 # formplayer_archive_time_spec: '10m'
 formplayer_purge_time_spec: '2d'
-formplayer_release_dir: "{{ formplayer_build_dir }}/formplayer__{{ new_release_name }}"
+formplayer_release_dir: >
+  {{ formplayer_build_dir }}/{{ "releases/" + new_release_name if new_release_name|default(None) else "current" }}
```

Comment:
Under what conditions will this evaluate to `{{ formplayer_build_dir }}/current`? Like when will `new_release_name` be blank or undefined? Is that expected to happen frequently?

Reply:
`new_release_name` will be null when this is being invoked from the `update-config` command: https://github.com/dimagi/commcare-cloud/pull/3253/files#diff-384a623aa676400249bff4d290514e36R224 (or whenever else we choose to have this file apply in-place to the live `current` directory instead of making a new release).

The question confirms my suspicion as I was writing this last night that maybe the this of `new_release_name` is not very clean. My overall purpose is to have this file be able to be run in two modes:
- the most common mode, in which a clean release is made and then current is symlinked to the release
- the in-place mode where files are updated directly in the current release (to maintain the current behavior of `update-config`).

Would love to hear your thoughts on how I can do that in a cleaner way

Reply:
> maybe the this of `new_release_name` is not very clean

I think you missed a word in there.

> Would love to hear your thoughts on how I can do that in a cleaner way

It's not clear to me if `update-config` copies the formplayer jar into the "current" dir? If yes, that seems like a bit more than a config update. Do you agree? Maybe that's part of the problem? If not...

Maybe add another variable. Something like
```yaml
formplayer_release_dir: "{{ formplayer_build_dir }}/{{ "releases/" + new_release_name }}"
formplayer_config_dir: "'{{ formplayer_build_dir }}/{{ "releases/" + new_release_name if new_release_name else "current" }}'"
```

Config files are written to the config dir, which sometimes points to "current". Release files (the jar) are always written to the release dir (and this does not happen on `update-config`).

I don't know if that's much clearer. There are a fair amount of moving parts here (different things happen in different run modes), and that complexity makes it hard to reason about.

Reply:
Yeah `update-config` definitely shouldn't and I'm pretty sure doesn't download/update/touch  `formplayer.jar`. It runs this with `--tags=localsettings` so only tasks with that tag (which is just the two that update the config files `application.properties` and `sentry.properties`) run.

OK I like where you're suggestion is going. I think I'll also stop inlining the "/current" postfix everywhere and make a variable for that as well. Hopefully what I end up with is a bit more readable